bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rfile.py

```diff
@@ -0,0 +1,156 @@
+# Author: Giacomo Parolini CERN 04/2025
+
+################################################################################
+# Copyright (C) 1995-2022, Rene Brun and Fons Rademakers.                      #
+# All rights reserved.                                                         #
+#                                                                              #
+# For the licensing terms see $ROOTSYS/LICENSE.                                #
+# For the list of contributors see $ROOTSYS/README/CREDITS.                    #
+################################################################################
+
+r"""
+\pythondoc RFile
+
+TODO: document RFile
+
+\code{.py}
+# TODO code example
+\endcode
+
+\endpythondoc
+"""
+
+from . import pythonization
+
+
+class _RFile_Get:
+    """
+    Allow access to objects through the method Get().
+    This is pythonized to allow Get() to be called both with and without a template argument.
+    """
+
+    def __init__(self, rfile):
+        self._rfile = rfile
+
+    def __call__(self, namecycle):
+        """
+        Non-templated Get()
+        """
+        import ROOT
+        import cppyy
+
+        key = self._rfile.GetKeyInfo(namecycle)
+        if key:
+            obj = ROOT.Experimental.Internal.GetRFileObjectFromKey(self._rfile, key)
+            return cppyy.bind_object(obj, key.GetClassName())
+        # No key
+        return None
+
+    def __getitem__(self, template_arg):
+        """
+        Templated Get()
+        """
+
+        def getitem_wrapper(namecycle):
+            obj = self._rfile._OriginalGet[template_arg](namecycle)
+            return obj if obj else None
+
+        return getitem_wrapper
+
+
+class _RFile_Put:
+    """
+    Allow writing objects through the method Put().
+    This is pythonized to allow Put() to be called both with and without a template argument.
+    """
+
+    def __init__(self, rfile):
+        self._rfile = rfile
+
+    def __call__(self, name, obj):
+        """
+        Non-templated Put()
+        """
+        className = type(obj).__cpp_name__
```

Comment:
Maybe add here a touch of sanitization incase the passed object does not have `__cpp_name__`

Reply:
A case that I guess would deserve its own unittest